config/initializers/cancancan.rb

```diff
@@ -0,0 +1,12 @@
+module CanCanUnauthorizedMessage
+  # Fix deprecated syntax calling I18n#translate (using keyword args) without using **
+  def unauthorized_message(action, subject)
+    keys = unauthorized_message_keys(action, subject)
+    variables = {:action => action.to_s}
+    variables[:subject] = (subject.class == Class ? subject : subject.class).to_s.underscore.humanize.downcase
+    message = I18n.translate(nil, **variables.merge(:scope => :unauthorized, :default => keys + [""]))
+    message.blank? ? nil : message
+  end
+end
+
+CanCan::Ability.prepend(CanCanUnauthorizedMessage)
```

Comment:
> The version of CanCan which fixes this is not currently compatible with OFN...

Do we need an issue for that? Patching here is like maintaining our own fork and not sustainable long term.

Reply:
It's tricky because we can't currently bump to the version which fixes this keyword arguments syntax deprecation, as it breaks other things in the build. I explored that option first (just upgrading the `cancancan` gem).

Reply:
Yeah, I believe that it's tricky but are you saying that we should never do it and just maintain this patch? Or do you think that it's worth upgrading one day?

I guess that a dependency we can't upgrade will be a problem one day anyway.

Reply:
We should definitely upgrade it when we can, but it'll take some work. This patch should be easy to drop when we get to that point.